perception/map_based_prediction/src/map_based_prediction_node.cpp

```diff
@@ -1864,16 +1859,10 @@ ManeuverProbability MapBasedPredictionNode::calculateManeuverProbability(
     constexpr float LC_PROB = 1.0;  // probability for left lane change
     constexpr float RC_PROB = 1.0;  // probability for right lane change
     lane_follow_probability = 0.0;
-    if (!left_paths.empty() && right_paths.empty()) {
-      left_lane_change_probability = LC_PROB;
-      right_lane_change_probability = 0.0;
-    } else if (left_paths.empty() && !right_paths.empty()) {
-      left_lane_change_probability = 0.0;
-      right_lane_change_probability = RC_PROB;
-    } else {
-      left_lane_change_probability = LC_PROB;
-      right_lane_change_probability = RC_PROB;
-    }
+
+    // If the given lane is empty, the probability goes to 0
+    left_lane_change_probability = LC_PROB * !left_paths.empty();
+    right_lane_change_probability = RC_PROB * !right_paths.empty();
```

Comment:
I prefer this way to multiplying different types.

```suggestion
    left_lane_change_probability = left_paths.empty() ? 0.0 : LC_PROB;
    right_lane_change_probability = right_paths.empty() ? 0.0 : RC_PROB;
```